docs/DeveloperGuide.md

```diff
@@ -1,34 +1,509 @@
-# Developer Guide
+# Developer Guide for PlanNUS
 
-## Design & implementation
+<table><tr><td><div style="text-align:center">
+    <img src="images/PlanNUSLogo.png" />
+</div></td></tr></table>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Before reading this document, you are recommended to read through the [user guide](https://ay2021s1-cs2113t-f12-1.github.io/tp/UserGuide.html).
 
+## Table of contents 
 
-## Product scope
-### Target user profile
+* Table of contents
+{:toc}
 
-{Describe the target user profile}
 
-### Value proposition
+## Setting up PlanNUS
 
-{Describe the value proposition: what problem does it solve?}
+First and foremost, the following steps are assuming that you already have a [GitHub](https://github.com) account set up beforehand. Once this has been done, proceed to __fork__ this [repo](https://github.com/AY2021S1-CS2113T-F12-1/tp), and __clone__ the fork into your computer using [Sourcetree](https://sourcetreeapp.com) or any other _Git GUI_.
 
-## User Stories
+The _IDE_ to be used should contain the latest version of _Java_ as this is the main programming language for this application. Thus you are highly recommended to use Intellij IDEA.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+The following are remaining steps to be taken to finish the set up:
 
-## Non-Functional Requirements
+1. Make sure that the version is configured as __JDK 11__.
+2. When prompted, __import__ the project as a __Gradle project__ (could take minutes to complete).
+3. Enter commands to ensure that PlanNUS functions as expected. You may refer to the _User Guide_ for valid commands.
 
-{Give non-functional requirements}
 
-## Glossary
 
-* *glossary item* - Definition
+## Design
 
-## Instructions for manual testing
+### Architecture
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/Architecture.png" alt="Architecture diagram of PlanNUS"/>
```

Comment:
Maybe you can consider explaining here why Apps is not interacting with other packages?
![image](https://user-images.githubusercontent.com/49813920/97535278-5eea6180-19f6-11eb-948a-0afb8c1ee069.png)